lib/pact/api/exec_command_request.ex

```diff
@@ -0,0 +1,259 @@
+    |> set_data(data)
+    |> set_code(code)
+    |> set_nonce(nonce)
+    |> set_metadata(meta_data)
+    |> add_keypairs(keypairs)
+    |> add_signers(signers)
+  end
+
+  def new(_opts), do: %__MODULE__{}
+
+  @impl true
+  def set_network(%__MODULE__{} = cmd_request, network) do
+    case NetworkID.new(network) do
+      %NetworkID{} = network_id -> %{cmd_request | network_id: network_id}
+      {:error, reason} -> {:error, [network_id: :invalid] ++ reason}
+    end
+  end
+
+  def set_network({:error, reason}, _network), do: {:error, reason}
+
+  @impl true
+  def set_data(%__MODULE__{} = cmd_request, data) do
+    case EnvData.new(data) do
+      %EnvData{} -> %{cmd_request | data: data}
+      error -> error
+    end
+  end
+
+  def set_data({:error, reason}, _data), do: {:error, reason}
+
+  @impl true
+  def set_code(%__MODULE__{} = cmd_request, code) when is_binary(code),
+    do: %{cmd_request | code: code}
+
+  def set_code(%__MODULE__{}, _code), do: {:error, [code: :not_a_string]}
+  def set_code({:error, reason}, _code), do: {:error, reason}
+
+  @impl true
+  def set_nonce(%__MODULE__{} = cmd_request, nonce) when is_binary(nonce),
+    do: %{cmd_request | nonce: nonce}
+
+  def set_nonce(%__MODULE__{}, _nonce), do: {:error, [nonce: :not_a_string]}
+  def set_nonce({:error, reason}, _nonce), do: {:error, reason}
+
+  @impl true
+  def set_metadata(%__MODULE__{} = cmd_request, %MetaData{} = meta_data),
+    do: %{cmd_request | meta_data: meta_data}
+
+  def set_metadata(%__MODULE__{} = cmd_request, meta_data) do
+    case MetaData.new(meta_data) do
+      %MetaData{} = meta_data -> %{cmd_request | meta_data: meta_data}
+      {:error, reason} -> {:error, [meta_data: :invalid] ++ reason}
+    end
+  end
+
+  def set_metadata({:error, reason}, _metadata), do: {:error, reason}
+
+  @impl true
+  def add_keypair(%__MODULE__{keypairs: keypairs} = cmd_request, %KeyPair{} = keypair),
+    do: %{cmd_request | keypairs: keypairs ++ [keypair]}
+
+  def add_keypair(%__MODULE__{} = cmd_request, keypair) do
+    case KeyPair.new(keypair) do
+      %KeyPair{} = keypair -> add_keypair(cmd_request, keypair)
+      {:error, reason} -> {:error, [keypair: :invalid] ++ reason}
+    end
+  end
+
+  def add_keypair({:error, reason}, _keypair), do: {:error, reason}
+
+  @impl true
+  def add_keypairs(%__MODULE__{} = cmd_request, []), do: cmd_request
+
+  def add_keypairs(%__MODULE__{} = cmd_request, [keypair | keypairs]) do
+    cmd_request
+    |> add_keypair(keypair)
+    |> add_keypairs(keypairs)
+  end
+
+  def add_keypairs(%__MODULE__{}, _keypairs), do: {:error, [keypairs: :not_a_list]}
+  def add_keypairs({:error, reason}, _keypairs), do: {:error, reason}
+
+  @impl true
+  def add_signer(%__MODULE__{signers: nil} = cmd_request, %Signer{} = signer),
+    do: %{cmd_request | signers: SignersList.new([signer])}
+
+  def add_signer(%__MODULE__{signers: signers} = cmd_request, %Signer{} = signer) do
+    %SignersList{signers: signers} = signers
```

Comment:
@EdwinGuayacan there are 3 variables assigned to **signers**, this could cause problems since variables in elixir are immutable. Please use proper variable names.